tools/bin/hawqregister

```diff
@@ -325,250 +326,221 @@ class HawqRegister(object):
-            except pg.DatabaseError as e:
-                print e
+        except pg.DatabaseError as e:
+            print e
+            sys.exit(1)
+        if ret:
+            self.failure_handler.commit(('SQL', query))
+        return ret
+
+    def check_database_encoding(self):
+        encoding_indx = self.accessor.get_database_encoding_indx(self.database)
+        encoding = self.accessor.get_database_encoding(encoding_indx)
+        if self.encoding.strip() != encoding:
+            logger.error('Database encoding from yaml configuration file(%s) is not consistent with encoding from input args(%s).' % (self.encoding, encoding))
+            sys.exit(1)
+
+    def check_policy_consistency(self):
+        policy = self.get_distribution_policy() # "" or "{1,3}"
+        if policy is None:
+            return
+        if self.distribution_policy == 'DISTRIBUTED RANDOMLY':
+            logger.error('Distribution policy of %s from yaml is not consistent with the policy of existing table.' % self.tablename)
+            self.failure_handler.rollback()
+            sys.exit(1)
+        tmp_dict = {}
+        for i, d in enumerate(self.schema):
+            tmp_dict[d['name']] = i + 1
+        # 'DISTRIBUETD BY (1,3)' -> {1,3}
+        cols = self.distribution_policy.strip().split()[-1].strip('(').strip(')').split(',')
+        original_policy = ','.join([str(tmp_dict[col]) for col in cols])
+        if policy.strip('{').strip('}') != original_policy:
+            logger.error('Distribution policy of %s from yaml file is not consistent with the policy of existing table.' % self.dst_table_name)
+            self.failure_handler.rollback()
+            sys.exit(1)
+
+    def set_yml_dataa(self, file_format, files, sizes, tablename, schema, distribution_policy, file_locations,\
+                      bucket_number, partitionby, partitions_constraint, partitions_name, partitions_compression_level,\
+                      partitions_compression_type, partitions_checksum, partitions_filepaths, partitions_filesizes, encoding):
+        self.file_format = file_format
+        self.files = files
+        self.sizes = sizes
+        self.src_table_name = tablename
+        self.schema = schema
+        self.distribution_policy = distribution_policy
+        self.file_locations = file_locations
+        self.bucket_number = bucket_number
+        self.partitionby = partitionby
+        self.partitions_constraint = partitions_constraint
+        self.partitions_name = partitions_name
+        self.partitions_compression_level = partitions_compression_level
+        self.partitions_compression_type = partitions_compression_type
+        self.partitions_checksum = partitions_checksum
+        self.partitions_filepaths = partitions_filepaths
+        self.partitions_filesizes = partitions_filesizes
+        self.encoding = encoding
+
+    def option_parser_yml(self, yml_file):
```

Comment:
-1 on this. Can you move the function outside? In my view, partition logic is not related to class `HawqRegister`. Besides, can you move `_mapping_tablename_from_yml` logic outside of HawqRegister class?